src/Web/Controllers/BasketController.cs

```diff
@@ -256,126 +272,120 @@ private async Task<OrderStatus> ApproveOrRejectPurchase(string merchantRuleDecis
-                {
-                    Street1 = checkoutDetails.ShippingAddress1,
-                    Street2 = checkoutDetails.ShippingAddress2,
-                    City = checkoutDetails.City,
-                    State = checkoutDetails.State,
-                    ZipCode = checkoutDetails.ZipCode,
-                    Country = checkoutDetails.CountryRegion
-                }
+                Street1 = checkoutDetails.ShippingAddress1,
+                Street2 = checkoutDetails.ShippingAddress2,
+                City = checkoutDetails.City,
+                State = checkoutDetails.State,
+                ZipCode = checkoutDetails.ZipCode,
+                Country = checkoutDetails.CountryRegion
             };
 
-            var billingAddress = new PaymentInstrumentAddress
+            var billingAddress = new AddressDetails
             {
                 FirstName = checkoutDetails.FirstName,
                 LastName = checkoutDetails.LastName,
                 PhoneNumber = checkoutDetails.PhoneNumber,
-                BillingAddressDetails = new PurchaseEventAddress
-                {
-                    Street1 = checkoutDetails.BillingAddress1,
-                    Street2 = checkoutDetails.BillingAddress2,
-                    City = checkoutDetails.BillingCity,
-                    State = checkoutDetails.BillingState,
-                    ZipCode = checkoutDetails.BillingZipCode,
-                    Country = checkoutDetails.BillingCountryRegion
-                }
+                Street1 = checkoutDetails.BillingAddress1,
+                Street2 = checkoutDetails.BillingAddress2,
+                City = checkoutDetails.BillingCity,
+                State = checkoutDetails.BillingState,
+                ZipCode = checkoutDetails.BillingZipCode,
+                Country = checkoutDetails.BillingCountryRegion
             };
 
-            var device = new PurchaseDeviceContext
+            var device = new DeviceContext
             {
                 DeviceContextId = _contextAccessor.GetSessionId(),
                 IPAddress = _contextAccessor.HttpContext.Connection.RemoteIpAddress.ToString(),
-                DeviceContextDetails = new DeviceContext
-                {
-                    DeviceContextDC = checkoutDetails.FingerPrintingDC,
-                    Provider = DeviceContextProvider.DFPFINGERPRINTING.ToString()
-                }
+                DeviceContextDC = checkoutDetails.DeviceFingerPrinting.FingerPrintingDC,
+                Provider = DeviceContextProvider.DFPFingerPrinting.ToString()
             };
 
-            var productList = new List<PurchaseProduct>();
-            foreach (var item in basketViewModel.Items)
+            Func<string, string> getCategoryFromName = (productName) =>
             {
-                productList.Add(new PurchaseProduct
+                if (productName.Contains("mug", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
probably should stick to having braces, yeah?

Reply:
Sounds good.